### PR TITLE
[MODCON-127] - Implement ECS tenant soft deletion functionality

### PR DESCRIPTION
**Pull request by azizbekxm**
## Approach

https://issues.folio.org/browse/MODCON-127
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/consortia/client/UserTenantsClient.java [hunk 1]
```diff
@@ -12,4 +13,7 @@ public interface UserTenantsClient {
 
   @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
   void postUserTenant(@RequestBody UserTenant userTenant);
+
+  @DeleteMapping
+  void deleteUserTenants();
```
Comment (azizbekxm) [inline src/main/java/org/folio/consortia/client/UserTenantsClient.java:18]: https://issues.folio.org/browse/MODUSERS-411 this is being implementing in scope this ticket

File: src/main/java/org/folio/consortia/repository/UserTenantRepository.java [hunk 2]
```diff
@@ -14,6 +14,10 @@
 
 @Repository
 public interface UserTenantRepository extends JpaRepository<UserTenantEntity, UUID> {
+
+  @Query("SELECT ut FROM UserTenantEntity ut WHERE ut.tenant.isDeleted IS NULL OR ut.tenant.isDeleted = FALSE")
+  Page<UserTenantEntity> getAll(Pageable pageable);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/consortia/repository/UserTenantRepository.java:19]: existsByTenantId method now is not used any more, so can be deleted

File: src/main/java/org/folio/consortia/repository/UserTenantRepository.java [hunk 3]
```diff
@@ -14,6 +14,10 @@
...
+
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/consortia/repository/UserTenantRepository.java:20]: Seems we need more changes in this repository.
We have SelfController that is called after login and provide available affiliations for user.
image
We should not show affiliations for the deleted tenants in Switch affiliation popup.

In UserTenantController we have method
getUserTenants(UUID consortiumId, UUID userId, String username, String tenantId, Integer offset, Integer limit)
that is used 
I think to be consistent we should apply the same filtering for methods getByUsernameAndTenantId() and get() as well as we going to do for method getByUserId() in scope of SelfController changes

File: src/main/java/org/folio/consortia/repository/UserTenantRepository.java (hunk 2 above)
Comment (azizbekxm) [reply src/main/java/org/folio/consortia/repository/UserTenantRepository.java:19]: Thanks removed

File: src/main/java/org/folio/consortia/repository/UserTenantRepository.java (hunk 3 above)
Comment (azizbekxm) [reply src/main/java/org/folio/consortia/repository/UserTenantRepository.java:20]: Applied Filter for GetByUserId(), getByUsernameAndTenantId()

**Review by SerhiiNosko — APPROVED** [review]

**Review by Dmitriy-Butramyou — APPROVED** [review]

**Pull request merged**